project/Build.scala

```diff
@@ -705,7 +734,7 @@ object Build {
 
   val commonIrProjectSettings = Def.settings(
       commonSettings,
-      publishSettings,
+      publishSettings(VersionScheme.BreakOnMinor),
```

Comment:
https://github.com/sbt/sbt/discussions/6572 revealed that using `BreakOnMinor` for `ir`, although technically correct, is going to destroy practical scenarios of dependencies. It won't be possible to resolve a conflict between sbt-scalajs 1.6.x and 1.7.x, although sbt-scalajs has `semver-spec`, because they depend on scalajs-ir, which is `pvp`. And that, even though the things that depend on sbt-scalajs do not directly reference scalajs-ir.

There are two workarounds:
* Declare `semver-spec` (`BreakOnSevere`, like sbt-scalajs): allows correct scenarios, but also silently lets through situations where two dependencies actually directly depend on conflicting versions of scalajs-ir.
* *Do not set* `versionScheme` at all for scalajs-ir: it allows correct scenarios (but sometimes with a spurious warning), and warns (but does not fail) for actually conflicting scenarios.

I don't know yet which choice is best.